flink/v2.0/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSinkExtended.java

```diff
@@ -134,7 +134,7 @@ public void testTwoSinksInDisjointedDAG() throws Exception {
 
     List<Row> leftRows = createRows("left-");
     DataStream<Row> leftStream =
-        env.fromCollection(leftRows, ROW_TYPE_INFO)
+        env.addSource(createBoundedSource(leftRows), ROW_TYPE_INFO)
```

Comment:
A bit of context: `createdBoundedSource` uses a special source which emits the records, then waits for the checkpoint to get triggered before shutting down. See https://github.com/apache/iceberg/blob/36d9c1b3920434c977d205decf174185de725edb/flink/v2.0/flink/src/test/java/org/apache/iceberg/flink/source/BoundedTestSource.java#L93.